python/feathub/feature_views/tests/test_feature_view.py

```diff
@@ -201,3 +202,80 @@ def test_duplicate_feature_names_throw_exception(self):
                 keep_source_fields=True,
             )
         self.assertIn("contains duplicated feature name", ctx.exception.args[0])
+
+    def test_get_output_fields(self):
+        field_names = ["id", "val1", "val2", "val3", "time"]
+        source = DataGenSource(
+            name="source_1",
+            schema=Schema(field_names, [Int64, Int64, Int64, Int64, String]),
+            timestamp_field="time",
+            timestamp_format="%Y-%m-%d %H:%M:%S",
+            keys=["id"],
+        )
+
+        feature_1 = Feature(
+            name="derived_val_1", dtype=types.Int64, transform="val1 + 1", keys=["val1"]
+        )
+
+        feature_2 = Feature(
+            name="derived_val_2", dtype=types.Int64, transform="val2 + 1", keys=["val2"]
+        )
+
+        feature_view_1 = DerivedFeatureView(
+            name="feature_view_1",
+            source=source,
+            features=["time", feature_1, feature_2],
+            keep_source_fields=True,
+        )
+
+        built_feature_view_1 = feature_view_1.build(self.registry)
```

Comment:
In `test_kafka_source_sink.py` there is a following code snippet:
```python
            flink_table_descriptor: NativeFlinkTableDescriptor = (
                from_descriptor.call_args[0][0]
            )
```

Similarly, we may also write the code here this way
```python
built_feature_view_1: FeatureView = feature_view_1.build(self.registry)
```

and remove the `raise Exception("FeatureView is expected.")` below.

Reply:
Note that the `if... else raise Exception` code block is to avoid `mypy` type check failure, seems the above code snippet doesn't solve the problem. I found that `mypy` can skip checking some lines if users make sure that the lines are correct by adding a note like `# type: ignore`. Do you think it is enough for a test?

Reply:
Thanks for the update. This solution is also acceptable to me.